tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,222 @@
 """Tests for ZHA config flow."""
-from unittest import mock
+
+import os
+
+import serial.tools.list_ports
+import zigpy.config
 
 from homeassistant.components.zha import config_flow
-from homeassistant.components.zha.core.const import CONTROLLER, DOMAIN, ZHA_GW_RADIO
-import homeassistant.components.zha.core.registries
+from homeassistant.components.zha.core.const import CONF_RADIO_TYPE, CONTROLLER, DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_SOURCE
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
 
-import tests.async_mock
+from tests.async_mock import AsyncMock, MagicMock, patch, sentinel
 from tests.common import MockConfigEntry
 
 
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo()
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1234"
+    port.description = "Some serial port"
+
+    return port
+
+
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[com_port()]))
+@patch.object(
+    config_flow,
+    "detect_radios",
+    AsyncMock(return_value={CONF_RADIO_TYPE: "test_radio"}),
```

Comment:
If we use `patch` it will detect that the target is a coroutine and use an `AsyncMock` by default as `new`. Then we can skip that and just set `return_value`.

I think `patch.object` will work too, but I haven't tested it.

Reply:
primarily I just didn't want to add the mock into test parameters, but I can add for detect and check that it was called with expected parameters.

Reply:
For this one, I'm using the mock so let it to be passed as a test parameter.